### PR TITLE
Notifications config

### PR DESCRIPTION
**Pull request by thelonelyvulpes**

### CODE REVIEW

File: Neo4j.Driver/Neo4j.Driver/Config.cs [hunk 1]
```diff
@@ -182,4 +183,6 @@ public long FetchSize
     /// major version and y is the minor version.
     /// </summary>
     public string UserAgent { get; set; } = ConnectionSettings.DefaultUserAgent;
+
+    public INotificationsConfig NotificationsConfig { get; internal set; }
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Config.cs:200]: Could do with a summary.

File: Neo4j.Driver/Neo4j.Driver/Internal/AsyncSession.cs [hunk 2]
```diff
@@ -84,6 +85,11 @@ bool reactive
             LastBookmarks = Bookmarks.From(config.Bookmarks);
             _initialBookmarks = LastBookmarks;
         }
+
+        if (config.NotificationsConfig != null)
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/AsyncSession.cs]: Could use null-coalescing operator here. Although changes the style from existing lines. Just a thought, no real preference from me.

File: Neo4j.Driver/Neo4j.Driver/Internal/Connector/IConnection.cs [hunk 3]
```diff
@@ -64,16 +64,21 @@ internal interface IConnection : IConnectionDetails, IConnectionRunner
...
     Task LogoutAsync();
 
     Task<IReadOnlyDictionary<string, object>> GetRoutingTableAsync(
         string database,
         string impersonatedUser,
         Bookmarks bookmarks);
 
-    Task<IResultCursor> RunInAutoCommitTransactionAsync(AutoCommitParams autoCommitParams);
-    Task BeginTransactionAsync(string database, Bookmarks bookmarks, TransactionConfig config, string impersonatedUser);
+    Task<IResultCursor> RunInAutoCommitTransactionAsync(AutoCommitParams autoCommitParams,
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/Connector/IConnection.cs]: Style suggestion. Make multiline declarations to match the ones above.

File: Neo4j.Driver/Neo4j.Driver/Internal/MessageHandling/Metadata/NotificationsCollector.cs [hunk 4]
```diff
@@ -60,16 +60,15 @@ private static INotification CollectNotification(IDictionary<string, object> not
...
         var foundOffset = posValue.TryGetValue("offset", 0L, out var offset);
         var foundLine = posValue.TryGetValue("line", 0L, out var line);
         var foundColumn = posValue.TryGetValue("column", 0L, out var column);
 
-        if (foundOffset || foundLine || foundColumn)
-        {
-            var position = new InputPosition((int)offset, (int)line, (int)column);
-            return new Notification(code, title, description, position, severity);
-        }
+        var position = foundOffset || foundLine || foundColumn
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/MessageHandling/Metadata/NotificationsCollector.cs]: Nice tidy up.

File: Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolMessageFactory.cs [hunk 5]
```diff
@@ -103,20 +115,32 @@ public HelloMessage NewHelloMessage(IConnection connection, string userAgent, IA
             connection.RoutingContext);
     }
 
+    public HelloMessage NewAuthlessHelloMessage(IConnection connection, string userAgent, INotificationsConfig notificationsConfig)
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolMessageFactory.cs]: Multiline declaration params to be consistent in class.

**Review by AndyHeap-NeoTech — APPROVED** [review]
Looks good. The classic driver feature that requires something to be passed down through x number of layers :) 

Just a couple of comments on minor style things where it is slightly inconsistent within a class.

File: Neo4j.Driver/Neo4j.Driver/Config.cs (hunk 1 above)
Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/Config.cs:200]: Does that cover it ?

File: Neo4j.Driver/Neo4j.Driver/Internal/AsyncSession.cs (hunk 2 above)
Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/Internal/AsyncSession.cs]: the null check was pointless so removed.

File: Neo4j.Driver/Neo4j.Driver.Tests.TestBackend/Protocol/Driver/NewDriver.cs [hunk 6]
```diff
@@ -141,8 +141,27 @@ List<string> GetPaths()
             configBuilder.WithFetchSize(data.fetchSize.Value);
         }
 
-        var logger = new SimpleLogger();
+        if (data.notificationsMinSeverity != null || data.notificationsDisabledCategories != null)
+        {
+            if (data.notificationsMinSeverity == "OFF")
```
Comment (RichardIrons-neo4j) [inline Neo4j.Driver/Neo4j.Driver.Tests.TestBackend/Protocol/Driver/NewDriver.cs]: Can the magic strings be derived from somewhere/defined as constants?

**Comment by RichardIrons-neo4j** [comment]
Just a small thing about magic strings. All looks good.

**Pull request merged**